# Indiviodual Plan

gitlink: https://github.com/zyl200708-sketch/dsci-100-project-eric.git

## load package and importing data

In [1]:
library(tidyverse)
url = 'https://drive.google.com/uc?export=download&id=1hhtMiO5tqLPAIPATESCbF4wFB3HGy7DQ'
players <- read_csv(url)
head(players)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


experience subscribe
1 Pro         TRUE    
2 Veteran     TRUE    
3 Veteran    FALSE    
4 Amateur     TRUE    
5 Regular     TRUE    
6 Amateur     TRUE    
  hashedEmail                                                      played_hours
1 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
2 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9  3.8        
3 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28  0.0        
4 23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5  0.7        
5 7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e  0.1        
6 f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977  0.0        
  name      gender Age
1 Morgan    Male    9 
2 Christian Male   17 
3 Blake     Male   17 
4 Flora     Female 21 
5 Kylie     Male   21 
6 Adrian    Female 17

In [16]:
distinct(players,experience)
distinct(players,subscribe)
distinct(players,gender)
distinct(players,Age)
players |> summarize(mean_played_hours = mean(played_hours, na.rm = TRUE),
                        mean_age = mean(Age, na.rm = TRUE))

experience
1 Pro       
2 Veteran   
3 Amateur   
4 Regular   
5 Beginner

subscribe
1  TRUE    
2 FALSE

gender           
1 Male             
2 Female           
3 Non-binary       
4 Prefer not to say
5 Agender          
6 Two-Spirited     
7 Other

Age
1   9 
2  17 
3  21 
4  19 
5  47 
6  22 
7  23 
8  25 
9  58 
10 28 
11 18 
12 33 
13 26 
14 20 
15 24 
16 27 
17 42 
18 11 
19 10 
20 38 
21 14 
22 15 
23 37 
24 16 
25 12 
26 46 
27 32 
28 49 
29 45 
30 NA 
31 50 
32 44 
33 57

mean_played_hours mean_age
1 5.845918          21.13918

In [14]:
url_sessions = "https://drive.google.com/uc?export=download&id=1qoiGA-idlIr7LE2972cbrWjOH2ar-88b"
sessions <- read_csv(url_sessions)
head(sessions)
dim(sessions)

Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


hashedEmail                                                     
1 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
4 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
5 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
6 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
  start_time       end_time         original_start_time original_end_time
1 30/06/2024 18:12 30/06/2024 18:24 1.71977e+12         1.71977e+12      
2 17/06/2024 23:33 17/06/2024 23:46 1.71867e+12         1.71867e+12      
3 25/07/2024 17:34 25/07/2024 17:57 1.72193e+12         1.72193e+12      
4 25/07/2024 03:22 25/07/2024 03:58 1.72188e+12         1.72188e+12      
5 25/05/2024 16:01 25/05/2024 16:12 1.71665e+12         1.71665e+12      
6 23/06/2024 15:08 23/06/2024 17:10 1.71916e+12         1.71916e+12

[1] 1535    5

# Data description

## Players

#### It has 196 obervations and 7 variables

+ `experience`: character, experience level of players, such as: `Pro`,`Veteran`,`Amateur`,`Regular`
+ `subscribe`: logical, whether or not player has subscribed, such as: `True` and `False`
+ `hashedEmail`: character, hashed email addrsses of players
+ `played_hours`: double, the game playing in hours 
+ `name`: character, names of players
+ `gender`: character, gender of players, as: `Male`,`Female`,`Non-binary`.`Prefer not to say`,`Agender`,`Two-SpiritedandOther`
+ `Age`: double, age of players

## Sessions

#### It has 1535 obervations and 5 variables

+ `hashedEmail`: character, hashed email addrsses of players
+ `start_time`: character, start time of the lastest game
+ `end_time`: character, end time of the lastest game
+ `original_start_time`: double, original data of start_time, not human readable
+ `original_end_time`: double, original data of end_time, not human readable

## Potential Issues

+ `HashedEmail` is a little bit useless in the data
+ The number of `Non-binary`, `Prefer not to say Agender`, `Two-Spirited` and `Other`are really small compare to `male` and `female`.
